doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,530 @@
+discarded. This behaviour can be customized, see the section
+:ref:`"Configuring mid-circuit measurements" <mcm_config>`.
+
+.. note::
+
+    Currently, postselection is only supported on :class:`~.pennylane.devices.DefaultQubit`.
+
+Conditional operators / Dynamic circuits
+----------------------------------------
+
+Users can create conditional operators controlled on mid-circuit measurements using
+:func:`~.pennylane.cond`. The condition for a conditional operator may simply be
+the measured value returned by a ``measure()`` call, or we may construct a boolean
+condition based on such values and pass it to ``cond()``:
+
+.. code-block:: python
+
+    @qml.qnode(dev)
+    @qml.defer_measurements
+    def qnode_conditional_op_on_zero(x, y):
+        qml.RY(x, wires=0)
+        qml.CNOT(wires=[0, 1])
+        m_0 = qml.measure(1)
+
+        qml.cond(m_0 == 0, qml.RY)(y, wires=0)
+        return qml.probs(wires=[0])
+
+    pars = np.array([0.643, 0.246], requires_grad=True)
+
+.. code-block:: pycon
+
+    >>> qnode_conditional_op_on_zero(*pars)
+    tensor([0.88660045, 0.11339955], requires_grad=True)
+
+For more examples, refer to the :func:`~.cond` documentation
+and the `how-to on creating dynamic circuits with mid-circuit measurements
+<https://pennylane.ai/qml/demos/tutorial_how_to_create_dynamic_mcm_circuits/>`__.
+
+.. _mid_circuit_measurements_statistics:
+
+Mid-circuit measurement statistics
+----------------------------------
+
+Statistics of mid-circuit measurements can be collected along with terminal measurement statistics.
+Currently, :func:`~.counts`, :func:`~.expval`, :func:`~.probs`, :func:`~.sample`, and :func:`~.var`
+are supported, and devices that currently support collecting such
+statistics are :class:`~.pennylane.devices.DefaultQubit`, :class:`~.DefaultMixed`,
+and :class:`~.DefaultQubitLegacy`.
+
+.. code-block:: python3
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def func(x, y):
+        qml.RX(x, wires=0)
+        m_0 = qml.measure(0)
+        qml.cond(m_0, qml.RY)(y, wires=1)
+        return qml.probs(wires=1), qml.probs(op=m_0)
+
+Executing this ``QNode``:
+
+.. code-block:: pycon
+
+    >>> func(np.pi / 2, np.pi / 4)
+    (tensor([0.9267767, 0.0732233], requires_grad=True),
+     tensor([0.5, 0.5], requires_grad=True))
+
+Users can also collect statistics on mid-circuit measurements manipulated using arithmetic/boolean operators.
+This works for both unary and binary operators. To see a full list of supported operators, refer to the
+:func:`~.pennylane.measure` documentation. An example for collecting such statistics is shown below:
+
+.. code-block:: python3
+
+    import pennylane as qml
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def circuit(phi, theta):
+        qml.RX(phi, wires=0)
+        m_0 = qml.measure(wires=0)
+        qml.RY(theta, wires=1)
+        m_1 = qml.measure(wires=1)
+        return qml.sample(~m_0 - 2 * m_1)
```

Comment:
Agree! Something to keep in mind, we should always try and keep quickstarts as concise as possible, and direct readers to relevant docstrings for more details. For a couple of reasons:

- Readers often treat quickstarts like a 'map', using it to get a lay of the land to find the functionality they need. And maps are easier to read if they are concise and provide an overview of what's available.

- It avoids duplicating documentation far from the docstring/where the code lives. It can be easy to remember to update a docstring in a PR, but forgetting to update the quickstart.

- docstrings tend to show up first in google, so should usually be the main source of truth

Of course, sometimes there isn't a specific docstring that makes sense for some information --- e.g., documentation about how simulation methods apply, etc. So this fits perfectly at the bottom of a quickstart

Reply:
I'd just add a more complex example to the `cond` docstring, then? :)

Reply:
I think that makes sense. Thanks for the insight @josh146 !